### PR TITLE
dispose of client should only flush

### LINKED ISSUE #599: Make Dispose only Flush
The SDK is designed to run for the lifetime of the app. Dispose as in making it unusable is only a source of problems (ASP.NET Core creates multiple containers and we rely on a resolve of `IHub` to init the SDK, but when the (first) container is disposed, it disposes also that Hub so if an exception is generated in between that hub being disposed and the new container being created/hub taking over, those errors are lost.

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #599

### CODE REVIEW

**Review by bruno-garcia — COMMENTED** [review]
I wonder if we need this also on `Hub.cs`

**Comment by SimonCropp** [comment]
@bruno-garcia does the hub exist in the container? and hence has the same problem?

**Comment by bruno-garcia** [comment]
Yes. Both are registered here:

https://github.com/getsentry/sentry-dotnet/blob/1605e97c8b7d9bd952bc1fc63ef45e654b6478fe/src/Sentry.Extensions.Logging/Extensions/DependencyInjection/ServiceCollectionExtensions.cs#L26-L27

File: test/Sentry.Tests/SentryClientTests.cs [hunk 1]
```diff
@@ -557,34 +574,36 @@ public void CaptureTransaction_NotFinished_Sent()
...
+    public void CaptureTransaction_DisposedClient_DoesNotThrow()
     {
         var sut = _fixture.GetSut();
         sut.Dispose();
-        _ = Assert.Throws<ObjectDisposedException>(() => sut.CaptureTransaction(null));
+        sut.CaptureTransaction(
+            new Transaction(
+                "test name",
+                "test operation"
+            )
```
Comment (lucas-zimerman) [inline test/Sentry.Tests/SentryClientTests.cs:585]:
```suggestion
                "test operation")
```

**Comment by lucas-zimerman** [comment]
Test is complaining about this: `Sentry.Tests.ApiApprovalTests.Run: VerifyException : Results do not match.`

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 
+Dispose of client should only flush ([#1354](https://github.com/getsentry/sentry-dotnet/pull/1354))
```
Comment (lucas-zimerman) [inline CHANGELOG.md:5]:
```suggestion
Dispose of client should only flush ([#1354](https://github.com/getsentry/sentry-dotnet/pull/1354))

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
...
 ### Fixes
```
Comment (lucas-zimerman) [inline CHANGELOG.md:7]:
```suggestion
### Fixes

Dispose of client should only flush ([#1354](https://github.com/getsentry/sentry-dotnet/pull/1354))
```

**Comment by SimonCropp** [comment]
@bruno-garcia so what do we do in hub.dispose?

        public void Dispose()
        {
            _options.LogInfo("Disposing the Hub.");

            if (Interlocked.Exchange(ref _isEnabled, 0) != 1)
            {
                return;
            }

            if (_integrations?.Length > 0)
            {
                foreach (var integration in _integrations)
                {
                    if (integration is IInternalSdkIntegration internalIntegration)
                    {
                        internalIntegration.Unregister(this);
                    }
                }
            }

            (_ownedClient as IDisposable)?.Dispose();
            _rootScope.Dispose();
            ScopeManager.Dispose();
        }

just return?

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@
 - Add CaptureLastError as an extension method to the Server class on ASP.NET ([#1411](https://github.com/getsentry/sentry-dotnet/pull/1411))
 - Add IsDynamicCode* to events ([#1418](https://github.com/getsentry/sentry-dotnet/pull/1418))
 
+### Fixes
```
Comment (bruno-garcia) [inline CHANGELOG.md:10]: Needs to be moved up

File: test/Sentry.Tests/SentryClientTests.cs [hunk 5]
```diff
@@ -419,13 +421,28 @@ public void CaptureUserFeedback_EventIdEmpty_FeedbackIgnored()
         //Assert
         _ = sut.Worker.DidNotReceive().EnqueueEnvelope(Arg.Any<Envelope>());
     }
+    [Fact]
```
Comment (bruno-garcia) [inline test/Sentry.Tests/SentryClientTests.cs:424]:
```suggestion

    [Fact]
```

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**